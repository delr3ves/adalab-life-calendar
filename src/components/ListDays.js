import React, { Component } from "react";
import PropTypes from "prop-types";
import { HAPPY } from "../core/MoodConstants";
import "../stylesheets/ListDays.scss";
import ReactTooltip from "react-tooltip";
import moodToEmoji from "./utils/MoodToEmoji";

const dayShape = PropTypes.shape({
  date: PropTypes.date,
  mood: PropTypes.string,
  comment: PropTypes.string
});

const SingleDay = props => {
  const { day } = props;
  const moodType = day.mood === HAPPY ? "success" : "warning";
  return (
    <div>
      <p data-tip data-for={day.date}>
        {moodToEmoji(day.mood)}
      </p>
      <ReactTooltip id={day.date} type={moodType} effect="solid">
        <h1>{day.date}</h1>
        <p>{day.comment}</p>
      </ReactTooltip>
    </div>
  );
};

SingleDay.propTypes = {
  day: dayShape
};

class ListDays extends Component {
  listOfDays(days) {
    return (
      <ul className="list_days__list">
        {days.map(day => (
          <li key={day.date} className="list_days__mood_item">
            <SingleDay day={day} />
          </li>
        ))}
      </ul>
    );
  }

  render() {
    const { days } = this.props;
    const list =
      days.length > 0 ? this.listOfDays(days) : <p>qué tal ha ido tu día</p>;
    return (
      <div className="list_days__container">
        <h1 className="list_days__title">Life - Calendar</h1>
        {list}
      </div>
    );
  }
}

ListDays.propTypes = {
  days: PropTypes.arrayOf(dayShape)
};

export default ListDays;
