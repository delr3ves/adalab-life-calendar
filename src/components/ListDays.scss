.list_days__container {
  width: 450px;
  height: 100vh;
  margin: 0 auto;
}

.list_days__title {
  font-size: 1.3em;
}

.list_days__list {
  list-style: none;
}

.list_days__mood_item {
  display: inline-block;
  margin: 5px;
  font-size: 3em;
  cursor: pointer;
}

