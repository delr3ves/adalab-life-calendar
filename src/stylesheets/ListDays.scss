.list_days__title {
  font-size: 1.3em;
}

.list_days__list {
  list-style: none;
  text-align: justify;
}

.list_days__mood_item {
  display: inline-block;
  margin: 15px;
  font-size: 5em;
  cursor: pointer;
  transition: opacity .3s ease-in-out;
  opacity: .8;
}

.list_days__mood_item:hover {
  opacity: 1;
}
