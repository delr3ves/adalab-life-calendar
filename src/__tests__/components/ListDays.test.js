import React from "react";
import renderer from "react-test-renderer";
import ListDays from "../../components/ListDays";

describe("ListDays component suite", () => {
  it("should show empty case when no days are found", () => {
    const emptyList = renderer.create(<ListDays days={[]} />).toJSON();
    expect(emptyList).toMatchSnapshot();
  });

  it("should list the emojis when we have some days", () => {
    const emptyList = renderer
      .create(
        <ListDays
          days={[
            {
              date: "2019-11-09",
              mood: "happy",
              comment: "yeeeeah"
            },
            {
              date: "2019-11-08",
              mood: "sad"
            }
          ]}
        />
      )
      .toJSON();
    expect(emptyList).toMatchSnapshot();
  });
});
