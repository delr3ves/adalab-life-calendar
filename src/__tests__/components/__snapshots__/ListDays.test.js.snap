// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListDays component suite should list the emojis when we have some days 1`] = `
<div
  className="list_days__container"
>
  <h1
    className="list_days__title"
  >
    Life - Calendar
  </h1>
  <ul
    className="list_days__list"
  >
    <li
      className="list_days__mood_item"
    >
      <div>
        <p
          data-for="2019-11-09"
          data-tip={true}
        >
          ðŸ˜Š
        </p>
        <div
          className="__react_component_tooltip place-top type-dark "
          data-id="tooltip"
          id="2019-11-09"
        >
          <h1>
            2019-11-09
          </h1>
          <p>
            yeeeeah
          </p>
        </div>
      </div>
    </li>
    <li
      className="list_days__mood_item"
    >
      <div>
        <p
          data-for="2019-11-08"
          data-tip={true}
        >
          ðŸ˜ª
        </p>
        <div
          className="__react_component_tooltip place-top type-dark "
          data-id="tooltip"
          id="2019-11-08"
        >
          <h1>
            2019-11-08
          </h1>
          <p />
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`ListDays component suite should show empty case when no days are found 1`] = `
<div
  className="list_days__container"
>
  <h1
    className="list_days__title"
  >
    Life - Calendar
  </h1>
  <p>
    quÃ© tal ha ido tu dÃ­a
  </p>
</div>
`;
